dom/src/test/java/org/apache/james/mime4j/field/ContentDispositionFieldTest.java

```diff
@@ -106,6 +113,48 @@ public void testGetFilename() throws Exception {
         assertNull(f.getFilename());
     }
 
+    public void testNonAsciiFilename() throws MimeException {
+        ContentDispositionField f =
+                parse("Content-Disposition: attachment;"
+                        + "\nfilename*0=\"=?UTF-8?Q?3-2_FORPROSJEKT_2-Sheet_-_XXX_A_2_40_?="
+                        + "\n=?UTF-8?Q?\";"
+                        + "\nfilename*1=\"201_-_Fasader_nord=C3=B8st_og_nordvest.dwg?=\"");
+
+        assertEquals(getMessage(f),
+                "3-2 FORPROSJEKT 2-Sheet - XXX A 2 40 201 - Fasader nordøst og nordvest.dwg",
+                f.getFilename());
+    }
```

Comment:
Can we get this test too for the lenient version?

Reply:
sure thing, but are suggesting to move this test to LenientContentDispositionFieldTest or just replicate the same test in LenientContentDispositionFieldTest?

Reply:
Replicate in my opinion.

The best way to go in the future is to have a common "contract" for both the lenient and non lenient implementations but this clearly looks like another topic to me.

Reply:
yes, I agree with you